modules/Microsoft.Network/privateLinkServices/deploy.bicep

```diff
@@ -0,0 +1,102 @@
+@description('Required. Name of the private link service to create.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+@description('Optional. Tags to be applied on all resources/resource groups in this deployment.')
+param tags object = {}
+
+@description('Optional. The extended location of the load balancer.')
+param extendedLocation object = {}
+
+@description('Optional. The auto-approval list of the private link service.')
+param autoApproval object = {}
+
+@description('Optional. Whether the private link service is enabled for proxy protocol or not.')
```

Comment:
@kavishshivhare  we are not providing sufficient test coverage for some of the parameters: i.e. autoapproval, enableProxy, extended location, fqdns, visibility. 

Can you please: Validate the usage of these parameters, and also provide some guidance in the readme on the parameter usage section as it is unclear how they can be used. At least the ones that are object and array types.